infra/prompt-flow/create-prompt-flow.sh

```diff
@@ -0,0 +1,89 @@
+sed -i "s@<aisearch_index>@${search_index}@g" "$flow_dag_file"
+
+pfazure flow create --subscription "$subscription_id" --resource-group "$resource_group" \
+    --workspace-name "$aml_workspace" --flow "$flow_dir" --set type=chat
+
+echo "Creating model"
+az ml model create --file "${SCRIPTPATH}/model.yaml" --resource-group "$resource_group" --workspace-name "$aml_workspace" --set "name=$model_name"
+
+if az ml online-endpoint show  --resource-group "$resource_group"  --workspace-name "$aml_workspace" --name "$endpoint_name" > /dev/null 2>&1; then
+    echo "Updating endpoint"
+    az ml online-endpoint update --resource-group "$resource_group" --workspace-name "$aml_workspace" --file "${SCRIPTPATH}/endpoint.yaml" --set "name=$endpoint_name" --set auth_mode=key
+else
+    echo "Creating endpoint"
+    az ml online-endpoint create --resource-group "$resource_group" --workspace-name "$aml_workspace" --file "${SCRIPTPATH}/endpoint.yaml" --set "name=$endpoint_name" --set auth_mode=key
+fi
+
+endpoint_id=$(az ml online-endpoint show --resource-group "$resource_group" --workspace-name "$aml_workspace"  --name "$endpoint_name" --query "identity.principal_id" --output tsv)
+az role assignment create --role "Azure Machine Learning Workspace Connection Secrets Reader" --scope "/subscriptions/${subscription_id}/resourceGroups/${resource_group}/providers/Microsoft.MachineLearningServices/workspaces/${aml_workspace}" --assignee-object-id "$endpoint_id" --assignee-principal-type ServicePrincipal
+
+prt_config_override="deployment.subscription_id=${subscription_id},deployment.resource_group=${resource_group},deployment.workspace_name=${aml_workspace},deployment.endpoint_name=${endpoint_name},deployment.deployment_name=${deployment_name}"
+model_version=$(az ml model list --resource-group "$resource_group" --workspace-name "$aml_workspace" --name "$model_name" --query "[].version | max(@)" --output tsv)
+if az ml online-deployment show  --resource-group "$resource_group"  --workspace-name "$aml_workspace" --endpoint-name "$endpoint_name" --name "$deployment_name" > /dev/null 2>&1; then
+    echo "Updating deployment"
+    az ml online-deployment update --file "${SCRIPTPATH}/deployment.yaml" --resource-group "$resource_group" --workspace-name "$aml_workspace" --set "name=$deployment_name" --set "endpoint_name=$endpoint_name" --set "model=azureml:${model_name}:${model_version}" --set environment_variables={} --set "environment_variables.PRT_CONFIG_OVERRIDE=$prt_config_override" --set environment_variables.PROMPTFLOW_RUN_MODE=serving
+else
+    echo "Creating deployment"
+    az ml online-deployment create --file "${SCRIPTPATH}/deployment.yaml" --resource-group "$resource_group" --workspace-name "$aml_workspace" --all-traffic --set "name=$deployment_name" --set "endpoint_name=$endpoint_name" --set "model=azureml:${model_name}:${model_version}" --set "environment_variables.PRT_CONFIG_OVERRIDE=$prt_config_override"
+fi
```

Comment:
1. do we not need `environment_variables.PROMPTFLOW_RUN_MODE=serving` in the else block whilst new deployment is being created ?
2. similar question for `--all-traffic` - do we not need this option in deployment update ?

Reply:
1. No becuase when creating, it pulls it from the `deployment.yaml`. For some reason when running `update` I had to declare the environment variables block from scratch so I have to readd it `--set environment_variables={}`. Without this parameter it would not let me set environment vars :( 
2. It's not an option for update, only create.